jboss-eap-standalone-openshift/azuredeploy.json

```diff
@@ -0,0 +1,375 @@
+      "defaultValue": "Standard_D4_v3",
+      "allowedValues": [
+        "Standard_D4_v3",
+        "Standard_D8_v3",
+        "Standard_DS4_v2",
+        "Standard_DS4_v3",
+        "Standard_DS8_v3",
+        "Standard_DS16_v3"
+      ],
+      "metadata": {
+        "description": "The size of the Virtual Machine."
+      }
+    },
+    "_artifactsLocation": {
+      "type": "string",
+      "metadata": {
+        "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+      },
+      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/jboss-eap-standalone-openshift"
+    },
+    "_artifactsLocationSasToken": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated."
+      },
+      "defaultValue": ""
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    }
+  },
+  "variables": {
+    "ScriptFolder": "scripts",
+    "ScriptFileName": "allinone.sh",
+    "gituser": "openshift",
+    "branch": "master",
+    "version": "3.7",
+    "location": "[resourceGroup().location]",
+    "virtualNetworkName": "openshiftVnet",
+    "addressPrefix": "10.0.0.0/16",
+    "vhdStorageType": "Premium_LRS",
+    "vnetId": "[resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworkName'))]",
+    "rhel": {
+      "publisher": "Redhat",
+      "offer": "RHEL",
+      "sku": "7-RAW",
+      "version": "latest"
+    },
+    "baseVMachineTemplateUriOneVm": "[concat(parameters('_artifactsLocation'), '/onevm.json')]",
+    "vmSizesMap": {
+      "storageAccountType": "Premium_LRS"
+    },
+    "Standard_D4_v3": {
+      "storageAccountType": "Premium_LRS"
+    },
+    "Standard_D8_v3": {
+      "storageAccountType": "Premium_LRS"
+    },
+    "Standard_DS4_v2": {
+      "storageAccountType": "Premium_LRS"
+    },
+    "Standard_DS4_v3": {
+      "storageAccountType": "Premium_LRS"
+    },
+    "Standard_DS8_v3": {
+      "storageAccountType": "Premium_LRS"
+    },
+    "Standard_DS16_v3": {
+      "storageAccountType": "Premium_LRS"
+    },
+    "singlequote": "'",
+    "apiVersion": "2015-06-15",
+    "apiVersionCompute": "2017-03-30",
+    "apiVersionNetwork": "2018-04-01",
+    "tmApiVersion": "2015-11-01",
+    "apiVersionStorage": "2018-07-01",
+    "apiVersionLinkTemplate": "2015-01-01",
+    "nicName": "OneVmNic",
+    "publicIPAddressName": "onevmPublicIP",
+    "publicIPAddressType": "Dynamic",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('virtualNetworkName'))]",
```

Comment:
remove concat() and just use resourceId, e.g.

[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), 'default')]

Reply:
done

Reply:
I had to revert back to the original since I couldn't get your suggestion to work.

Reply:
look at some of the other samples in the repo - it's possible you're overloading the subnet name with vnetname - if so, add a variable for the subnet name